### ISSUE TITLE
default keyword does not generate same IL code as default(T) in case of nullable argument

### ISSUE THREAD

**Issue by Drake103** [dotnet/roslyn#23340]
**Version Used**: 
C# 7.1

**Steps to Reproduce**:

1.  Create function with `bool? someFlag = default` parameter.

**Expected Behavior**:
Parameter is `null` when not passed (as for `default(bool?)`)

**Actual Behavior**:
Parameter is false by default.

Sample project:
[DefaultBehavior.zip](https://github.com/dotnet/roslyn/files/1492614/DefaultBehavior.zip)
---
`public void HandleDefaultBool(bool? flag = default(bool?))`
generates the following IL:
```
.method public hidebysig instance void 
    HandleDefaultBool(
      [opt] valuetype [System.Runtime]System.Nullable`1<bool> flag
    ) cil managed 
  {
    .param [1] = nullref
    .maxstack 2
    .locals init (
      [0] bool V_0
    )

    ...      

  } // end of method SampleClass::HandleDefaultBool
```

while `public void HandleDefault(bool? flag = default)` generates:
```
.method public hidebysig instance void 
    HandleDefault(
      [opt] valuetype [System.Runtime]System.Nullable`1<bool> flag
    ) cil managed 
  {
    .param [1] = bool(false)
    .maxstack 2
    .locals init (
      [0] bool V_0
    )

    ... 

  } // end of method SampleClass::HandleDefault
```

**Comment by HaloFour** [comment]
This is a bug and it should be fixed in the 7.2 release of the language:

#22578

**Comment by Drake103** [comment]
Thank you, somehow I could not find it using filter.
